server/handler/issue_comment.go

```diff
@@ -81,34 +74,34 @@ func (h *IssueComment) Handle(ctx context.Context, eventType, deliveryID string,
 		return err
 	}
 
-	fc := h.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	switch {
-	case fc.LoadError != nil || fc.ParseError != nil:
+	case evalCtx.Config.LoadError != nil || evalCtx.Config.ParseError != nil:
 		logger.Warn().Str(LogKeyAudit, "issue_comment").Msg("Skipping tampering check because the policy is not valid")
-	case fc.Config != nil:
-		tampered := h.detectAndLogTampering(ctx, prctx, client, event, fc.Config)
+	case evalCtx.Config.Config != nil:
+		tampered := h.detectAndLogTampering(ctx, evalCtx, event)
 		if tampered {
 			return nil
 		}
 	}
 
-	evaluator, err := h.Base.ValidateFetchedConfig(ctx, prctx, client, fc, common.TriggerComment)
+	evaluator, err := evalCtx.ParseConfig(ctx, common.TriggerComment)
 	if err != nil {
 		return err
 	}
 	if evaluator == nil {
 		return nil
 	}
 
-	result, err := h.Base.EvaluateFetchedConfig(ctx, prctx, client, evaluator, fc)
+	result, err := evalCtx.EvaluatePolicy(ctx, evaluator)
 	if err != nil {
 		return err
 	}
 
-	return h.Base.RequestReviewsForResult(ctx, prctx, client, common.TriggerComment, result)
+	evalCtx.RunPostEvaluateActions(ctx, result, common.TriggerComment)
```

Comment:
This fixes a bug from when review dismissal was added: we weren't dismissing reviews in response to comment events evaluations

Reply:
i thought that was intentional, because we're only looping through the [reviews](https://github.com/palantir/policy-bot/pull/476/files#diff-9a1aa4ae85323f98b833a9551817bf9388a01bfef333c49b28c08927f5512f2fR49), but that's a good point, maybe we should also do the same for comments too. I think we might need to change up the dismissal logic a bit to support it.

Reply:
I think you're right that this doesn't change behavior (no reviews should be dismissed in response to comment events), but that's also true of other events where dismissal was evaluated by virtue of being part of `Evaluate`.

My main goal is to make all the "post-review" actions consistent so it's harder to miss in future cases where it does matter. If we want to optimize review dismissal further, I think it should check if the `trigger` relates to reviews or commits, similar to what reviewer assignment does.